assets/app/scripts/services/data.js

```diff
@@ -339,7 +339,16 @@ angular.module('openshiftConsole')
   DataService.prototype.watch = function(type, context, callback, opts) {
     type = normalizeType(type);
     opts = opts || {};
-    this._watchCallbacks(type, context).add(callback);
+
+    if (callback) {
+      // If we were given a callback, add it
+      this._watchCallbacks(type, context).add(callback);
+    }
+    else if (!this._watchCallbacks(type, context).has()) {
+      // We can be called with no callback in order to re-run a list/watch sequence for existing callbacks
+      // If there are no existing callbacks, return
+      return {};
```

Comment:
I assume it's safe to later remove `undefined` from callbacks in `unwatch`. I didn't see the jQuery doc say one way or the other.